services/ui-src/src/utils/forms/schemas.ts

```diff
@@ -27,6 +35,46 @@ export const number = () =>
     .required(error.REQUIRED_GENERIC);
 export const numberOptional = () => number().notRequired();
 
+// Number - Ratio
+export const ratio = () =>
+  mixed()
+    .test({
+      message: error.INVALID_RATIO,
+      test: (val) => {
+        const replaceCharsRegex = /[,.:]/g;
+        const ratio = val.split(":");
+
+        // Double check and make sure that a ratio contains numbers on both sides
+        if (
+          ratio.length != 2 ||
+          ratio[0].trim().length == 0 ||
+          ratio[1].trim().length == 0
+        ) {
+          return false;
+        }
+
+        // Check if the left side of the ratio is a valid number
+        const firstTest = ignoreCharsForSchema(
+          ratio[0],
+          replaceCharsRegex
+        ).isValidSync(val);
+
+        // Check if the right side of the ratio is a valid number
+        const secondTest = ignoreCharsForSchema(
+          ratio[1],
+          replaceCharsRegex
+        ).isValidSync(val);
+
+        // If both sides are valid numbers, return true!
+        return firstTest && secondTest;
+      },
+    })
+    .test({
+      message: error.REQUIRED_GENERIC,
+      test: (val) => val != "",
+    })
+    .required(error.REQUIRED_GENERIC);
```

Comment:
Required message should fire when the field is empty instead of the ratio message. Just need to move these two above the first test. Might need to make same change for Number.

Reply:
Worked like a charm. Fixed!